File: 1500-count-largest-group/1500-count-largest-group.py
class Solution:
    def countLargestGroup(self, n: int) -> int:
        hashMap=collections.Counter()

        for i in range(1,n+1):
            key=sum(int(x) for x in str(i)) #We made key {sum of digit as key}
            hashMap[key] +=1        #Put values for Key {count of each digit sum}

        maxval=max(x for x in hashMap.values()) #Find Highest repeating num
        count=sum(1 for v in hashMap.values() if v==maxval) #Count freq of highest digit si,

        return count
        
        

File: 1500-count-largest-group/1500-count-largest-group.java
class Solution {
    public int countLargestGroup(int n) {
        int[] ds = new int[37];
        for (int i = 1; i <= n; i++) {
            ds[sum(i)]++;
        }
        int cnt = 1, maxval = 0;

        for (int i = 0; i < 37; i++) {
            if (ds[i] > maxval) {
                cnt=1;;
                maxval = ds[i];
            } else if (ds[i] == maxval){
                cnt++;
            }
        }
        return cnt;
    }

    public int sum(int n) {
        int ans = 0;
        while (n > 0) {
            ans += n % 10;
            n /= 10;
        }
        return ans;
    }
}
